pkg/schemadsl/compiler/translator.go

```diff
@@ -713,6 +793,17 @@ func translateSpecificTypeReference(tctx *translationContext, typeRefNode *dslNo
 		ref.RequiredExpiration = &core.ExpirationTrait{}
 	}
 
+	// Add the deprecation, if any.
+	if slices.Contains(tctx.enabledFlags, "deprecation") && slices.Contains(tctx.allowedFlags, "deprecation") {
```

Comment:
move this check into translateDeprecation and have it return an error if not enabled but specified